code/processes/gateway.q

```diff
@@ -413,12 +413,12 @@ getserverids:{[att]
   ];
 
   // its a dictionary of attributes
-  // check if attribute types are correct types
-  correctAttTypes:`dates`servertypes`tables!(14h;11h;11h);
-  w:(correctAttTypes [key att]=type each att);
-  if[not all{[x;y] y each key x}[att;w];
-  '("Wrong function parameter types provided.", " ",(raze " " sv string (where not w))," parameter(s) need type(s) ",raze " " sv string correctAttTypes where not w),"h"
-  ];
+    // check if attribute types are correct types
+    correctAttTypes:`dates`servertypes`tables!14 11 11h;
```

Comment:
The dictionary passed in will have `servertype rather than `servertypes, in the key so just drop the 's' here so it indexes correctly. Same for `dates I think.